packages/aws-cdk-lib/core/lib/annotations.ts

```diff
@@ -24,14 +24,59 @@ export class Annotations {
     this.stackTraces = !disableTrace;
   }
 
+  /**
+   * Acknowledge a warning. When a warning is acknowledged for a scope
+   * all warnings that match the id will be ignored.
+   *
+   * The acknowledgement will apply to all child scopes
+   *
+   * @example
+   * declare const stack: Stack;
+   * Annotations.of(stack).acknowledgeWarning('SomeWarningId', 'This warning can be ignored because...');
+   *
+   * @param id - the id of the warning message to acknowledge
+   * @param message optional message to explain the reason for acknowledgement
+   */
+  public acknowledgeWarning(id: string, message?: string): void {
+    const scopes = this.scope.node.findAll().map(child => child.node.path);
+    this.addMessage(cxschema.ArtifactMetadataEntryType.ACKNOWLEDGE, {
+      id,
+      message,
+      scopes,
+    });
+  }
+
   /**
    * Adds a warning metadata entry to this construct.
    *
    * The CLI will display the warning when an app is synthesized, or fail if run
    * in --strict mode.
    *
+   * @example
+   * declare const construct: Construct;
+   * Annotations.of(construct).addWarningV2('Library:Construct:ThisIsAWarning', 'Some message explaining the warning');
+   *
+   * @param id the unique identifier for the warning. This can be used to acknowledge the warning
    * @param message The warning message.
    */
+  public addWarningV2(id: string, message: string) {
+    const warning = {
+      id: id,
+      scope: this.scope.node.path,
```

Comment:
Metadata messages are already attached to a scope, so this information will be doubled?